src/preprocess.py

```diff
@@ -47,8 +51,46 @@ def load_corpus_from_binary(path: str) -> Dict[str, List[str]]:
     with open(path,"rb") as f:
         corpus = pkl.load(f)
     return corpus
+    
+def tokenize_document(document: str) -> List[str]:
+    """tokenize a document with nltk
+    
+    Arguments:
+        document {str} -- a str representing the document, query, etc..
+    
+    Returns:
+        List[str] -- lowered tokens
+    """
+    tokens = word_tokenize(document)
+    return [tok.lower() for tok in tokens]
+
+def tokenize_collection(collection: Dict[str, str]) -> Dict[str, List[str]]:
+    new_collection = OrderedDict()
+    for key in collection:
+        new_collection[key] = tokenize_document(collection[key])
+
+    return new_collection
+
+def load_stop_words(stop_word_path: str) -> List[str]:
+    with open(stop_word_path,"r") as f:
+        stp = [word.lower() for word in f.read().split("\n") if word != ""]
+    return stp
+
+def remove_stop_words_from_document(d: List[str], stop_words: List[str], exceptions: List[str] = []) -> List[str]:
```

Comment:
Because it depends on the model
-> Bool query : AND, OR, NAND are specials tokens (and we could add '(' and ')' )
-> Vectorial : AND , OR  are stop words